lib/cloudregister/registerutils.py

```diff
@@ -1407,6 +1407,19 @@ def write_framework_identifier(cfg):
         framework_file.write(json.dumps(identifier))
 
 
+# ----------------------------------------------------------------------------
+def instance_has_ip_enabled(rmt_ip):
+    """Check if the instance can make IPv4 or IPv6 requests."""
+    try:
+        socket.create_connection((rmt_ip, 443), timeout=2)
+    except OSError as e:
+        # check socket connection produced
+        # Network is unreachable error
+        return e.errno != errno.ENETUNREACH
```

Comment:
So if we get an access error for whatever reason we return `True`, i.e. indicating that we can reach the given IP address. I think any error on the socket creation should return `False` there is no reason to believe that if the error is access or resource limitation, i.e. no more sockets available the problem would go away by the time we need the network path.

Reply:
The goal is to check if the instance is able to make IPv4 or IPv6 requests, not if we can reach the RMT server (which coud be down or not available for different reasons)
Agree that no more sockets available should not be treated as `OK you can make $IP_format requests`

Reply:
Updated to return `False` to any socket error, I still think we could fine grain it more
as the purpose is only if the instance can make $IP_format requests

Reply:
Having the check for only that purpose would save us debugging and conversations and bugzilla entries when users start using IP4 and IPv6 combined or just IPv4

Reply:
At present the implementation is used to specifically check for being able to access the update server IP provided by the user on the command line. This will catch for example if the user provides an IPv6 address but the network is not configured for IPv6. It will also fail if the network is configured for IPv6 but the target server is down. Which is fine in this specific use case but would not be OK in the general use case where we want to know if we can reach addresses over IPv4 and/or IPv6 to potentially reduce the current probing that we do when processing the information we get from the region server. Both can be accomplished with the following implementation:


```
def has_ipv4_access():
   """Check if we have IPv4 network configuration"""
  return has_network_access_by_ipversion('8.8.8.8')

def has_ipv6_access():
   """Check if we have IPv6 network configuration"""
   return has_network_access_by_ipversion('2001:4860:4860::8888')

def has_network_access_by_ipversion(server_ip):
    """Check if we can connect to the given server"""
    try:
        connection = socket.create_connection((server_ip, 443), timeout=2)
    except OSError as e:
        logging.info('Network access error: "%s"', e)
        return False

    connection.close()
    return True
```

This makes the implementation of `has_network_access_by_ipversion` generic and can be used for the purposes of the command line argument verification where we want to know if a specific update server can be reached with the given IP address. And for the use case where we want to find out if IPv6 and/or IPv4 work we have the functions that use the Google DNS server to check if things work. The logging of a potential error message takes out the guess work why the connection may have failed.

Reply:
Updated the method for the logging message, the rest of the changes should be a new/different PR after merging this one